shell/browser/electron_browser_context.cc

```diff
@@ -123,8 +123,7 @@ ElectronBrowserContext::ElectronBrowserContext(const std::string& partition,
   base::StringToInt(command_line->GetSwitchValueASCII(switches::kDiskCacheSize),
                     &max_cache_size_);
 
-  CHECK(base::PathService::Get(chrome::DIR_USER_DATA, &path_));
-
+  base::PathService::Get(DIR_USER_CACHE, &path_);
```

Comment:
As explained in https://github.com/electron/electron/pull/19693#issuecomment-773998426, changing the default path of browser cache is a breaking change, since some apps will lose login session and even unsaved work.

And `userCache` is a very confusing name here, since not all users would consider browser cookies and localStorage as cache.

I still think introducing a new path key `browserData` is the only the solution that keeps compatibility with existing apps and does not cause confusions.

Reply:
Thanks I will review according to your comments. People doing the assumption we will change the 'cache' location on the fly at anytime or between 2 sessions. This is clearly no expected and does not make sense.
We already use this feature for more than 2 years on our side for fixing Citrix/Remote display disk space. It is part of our application to set the cache at a particular location at the installation, The cache location is set before the app 'ready' event and until now we never loose data. If for any reason, the user decides to change it, the behavior is like a clear cache option in any browsers, a warning explains the side effects.

Reply:
all your concerns are valid but already present if you play with the 'userData'. Confess, I did not improve so much the situation except trying to separate, user vs browser data folders.

Reply:
Hi, how can we progress ?
Except, if I did a mistake.
There is no breaking change because 
- DIR_USER_CACHE = DIR_CACHE + ApplicationName 
- DIR_CACHE = DIR_APP_DATA by default
- DIR_USER_DATA = DIR_APP_DATA + ApplicationName
so by default DIR_USER_CACHE = DIR_USER_DATA, this is backward compatible.

There is no regression because 
We can already play with DIR_APP_DATA ("appData") or DIR_USER_DATA ("userData") using app.setPath and then "lost" our browser data

Agree, this MR is a small step for moving to a clear separation between user/browser data but it has the advantage to offer a solution for all applications that needs/have to move browser data to another location (Citrix, Xen App, ...) and fix opened issues related to.

If DIR_CACHE ("cache") and DIR_USER_CACHE ("userCache") are not the correct naming and variables, no concern to use another ones, let me know.

Reply:
There are apps storing cache data in the `cache` dir, changing the default path of `cache` to `appData` is also a breaking change, with it some apps will start polluting users' `%APPDATA%` dir with cache data, in the worst case the cache data will overwrite user data.

Can you add a new `browserData` path instead? It also aligns with Chrome browser's name convention that "cache" usually only means cached files and cached that does not include user data.

<img width="325" alt="Screen Shot 2021-09-23 at 9 19 31" src="https://user-images.githubusercontent.com/639601/134438696-8471c1a9-aae6-4f8c-8525-b7cfa80fcddb.png">

<img width="528" alt="Screen Shot 2021-09-23 at 9 19 51" src="https://user-images.githubusercontent.com/639601/134438703-e31b256c-8ea8-46e9-a0ef-80d129f9b748.png">

Reply:
Understood, I will not change cache but add a browserdata path. Will work on that. 
"browserdata" will be equivalent to APP_DATA / ApplicationName by default. 
Could be convenient to have the equivalent of APP_DATA, 'browserappdata'(?!) which is the root folder for your browserdata.

Reply:
> Could be convenient to have the equivalent of APP_DATA, 'browserappdata'(?!) which is the root folder for your browserdata.

Hmm I can't think of a use case for it for now, we can add it in future if there are requests for it.